gap/HomStructure.gi

```diff
@@ -52,7 +55,7 @@ InstallMethodForCompilerForCAP( ExternalHomDiagram,
     
     morphisms := List( [ 1 .. nr_m ], mor_pair );
     
-    objects := Concatenation( sources, List( morphisms, m -> Range( m[1][2] ) ) );
+    objects := Concatenation( [ sources, List( morphisms, m -> Range( m[1][2] ) ) ] );
```

Comment:
Note: I'm planning to actually prefer the version with multiple arguments. But the conversion will happen automatically, so you are free to choose in the future.